File: src/main/java/org/cimsbioko/navconfig/Hierarchy.java
package org.cimsbioko.navconfig;

import java.util.List;
import java.util.Map;

public interface Hierarchy {

    String HOUSEHOLD = "household", INDIVIDUAL = "individual";

    Map<String, String> getLevelLabels();

    List<String> getAdminLevels();

    List<String> getLevels();

    String getParentLevel(String level);
}

File: src/main/java/org/cimsbioko/navconfig/Hierarchy.kt
package org.cimsbioko.navconfig

interface Hierarchy {

    val levelLabels: Map<String, String>
    val adminLevels: List<String>
    val levels: List<String>
    fun getParentLevel(level: String): String?

    companion object {
        const val HOUSEHOLD = "household"
        const val INDIVIDUAL = "individual"
    }
}